software/genmap/mappability/main.nf

```diff
@@ -0,0 +1,41 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process GENMAP_MAPPABILITY {
+    tag '$fasta'
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:[:], publish_by_meta:[]) }
+
+    conda (params.enable_conda ? "bioconda::genmap=1.3.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/genmap:1.3.0--h1b792b2_1"
+    } else {
+        container "quay.io/biocontainers/genmap:1.3.0--h1b792b2_1"
+    }
+
+    input:
+    path fasta
+
+    output:
+    path "*.wig"                  , emit: wig
+    path "*.version.txt"          , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+
+    """
+    genmap index -F $fasta -I genmap_index
+    genmap map \\
+        $options.args \\
+        -I genmap_index \\
+        -O . \\
+        -w
```

Comment:
```suggestion
```

Hmmm...maybe we should have optional outputs depending on whether `-w`/ `-bg` / `-t` are provided as opposed to hard-coding wig files as output?
https://github.com/cpockrandt/genmap#computing-the-mappability

We can check which options have been passed to the module using logic like this:
https://github.com/nf-core/modules/blob/4575e5455c62560e0cabdf128ec3adcc81f84a2c/software/rsem/preparereference/main.nf#L33

and then have optional outputs for each of these like here:
https://github.com/nf-core/modules/blob/4575e5455c62560e0cabdf128ec3adcc81f84a2c/software/star/align/main.nf#L34